### PR TITLE
Fix: missing fields in response

### LINKED ISSUE #24: Warnings about missing fields in ResourceCollector
lib/PaymentRails/ResourceCollector.php has warnings and errors about missing fields:

- Notice: Undefined index: meta in /lib/PaymentRails/ResourceCollection.php on line 43
- Notice: Trying to access array offset on value of type null in /lib/PaymentRails/ResourceCollection.php on line 43
- Notice: Undefined index: meta in /lib/PaymentRails/ResourceCollection.php on line 45
- Notice: Trying to access array offset on value of type null in /sdk-php/lib/PaymentRails/ResourceCollection.php on line 45

### PR DESCRIPTION
**Pull request by Aman-Aalam**
# Fixes #24 

This PR checks if the JSON response contains "meta" fields before adding them to response object, to prevent from PHP compilers from generating error messages.

### Checklist

- [x] I acknowledge that all my contributions will be made under the project's license
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have titled the PR appropriately
- [x] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [x] I have added inline documentation to the code I modified

If you have questions, create a GitHub Issue in this repository.

### CODE REVIEW

File: lib/PaymentRails/Version.php [hunk 1]
```diff
@@ -11,8 +11,8 @@ class Version
      * class constants
      */
     const MAJOR = 2;
-    const MINOR = 2;
-    const TINY = 0;
+    const MINOR = 1;
+    const TINY = 4;
```
Comment (suitedjimster) [inline lib/PaymentRails/Version.php:16]: shouldn't this be 2.2.1?

File: lib/PaymentRails/ResourceCollection.php [hunk 2]
```diff
@@ -40,9 +40,10 @@ class ResourceCollection implements Iterator
      */
     public function  __construct($response, $items, $pager)
     {
-        $this->_page = $response["meta"]["page"];
+        // Add "meta" fields only when they exist in JSON response
+        $this->_page = isset($response["meta"]) ? $response["meta"]["page"] : null;
         $this->_items = $items;
-        $this->_records = $response["meta"]["records"];
+        $this->_records = isset($response["meta"]) ? $response["meta"]["records"] : null;
```
Comment (Yongyiw) [inline lib/PaymentRails/ResourceCollection.php:46]: $response["meta"]["records"] ?? null

**Review by Yongyiw — APPROVED** [review]

File: lib/PaymentRails/ResourceCollection.php [hunk 3]
```diff
@@ -40,9 +40,10 @@ class ResourceCollection implements Iterator
      */
     public function  __construct($response, $items, $pager)
     {
-        $this->_page = $response["meta"]["page"];
+        // Add "meta" fields only when they exist in JSON response
+        $this->_page = isset($response["meta"]) ? $response["meta"]["page"] : null;
```
Comment (suitedjimster) [inline lib/PaymentRails/ResourceCollection.php:44]: which PHP version are we using?
if php7+, we can use null coalescing.`$response["meta"]["page"] ?? null`

File: lib/PaymentRails/ResourceCollection.php (hunk 2 above)
Comment (suitedjimster) [reply lib/PaymentRails/ResourceCollection.php:46]: @Yongyiw hmm.. according to our docs.. PHP requirement is 5.X... which doesn't support null coalescing..

**Review by suitedjimster — APPROVED** [review]

**Pull request merged**